File: backend/schemas/accounts.py
from pydantic import BaseModel, Field
from typing import List, Optional
from datetime import datetime
from models.accounts import AccountCategory

class AccountBase(BaseModel):
    code: str
    name: str
    category: AccountCategory

class AccountCreate(AccountBase):
    pass

class AccountUpdate(BaseModel):
    code: Optional[str] = None
    name: Optional[str] = None
    category: Optional[AccountCategory] = None

class AccountResponse(AccountBase):
    id: str
    is_active: bool
    is_system: bool
    current_balance: float = 0.0
    
    class Config:
        from_attributes = True

class DashboardStatsResponse(BaseModel):
    total_revenue: float
    total_expenses: float
    net_profit: float
    total_assets: float
    cash_balance: float
    bank_balance: float
    ar_balance: float
    ap_balance: float

class JournalEntryLineBase(BaseModel):
    account_id: str
    branch_id: Optional[str] = None
    warehouse_id: Optional[str] = None
    description: Optional[str] = None
    debit: float = Field(default=0.0, ge=0)
    credit: float = Field(default=0.0, ge=0)

class JournalEntryLineCreate(JournalEntryLineBase):
    pass

class JournalEntryLineResponse(JournalEntryLineBase):
    id: str
    
    class Config:
        from_attributes = True

class JournalEntryBase(BaseModel):
    reference: str
    description: str
    branch_id: Optional[str] = None
    source_module: Optional[str] = None
    source_id: Optional[str] = None
    date: Optional[datetime] = None

class JournalEntryCreate(JournalEntryBase):
    lines: List[JournalEntryLineCreate]

class JournalEntryResponse(JournalEntryBase):
    id: str
    status: str
    created_at: datetime
    lines: List[JournalEntryLineResponse]
    
    class Config:
        from_attributes = True

class LedgerRow(BaseModel):
    date: datetime
    reference: str
    journal_desc: Optional[str]
    line_desc: Optional[str]
    account_name: str
    debit: float
    credit: float
    balance: float
    status: Optional[str] = None
    source_id: Optional[str] = None
    created_by_name: Optional[str] = None

class LedgerResponse(BaseModel):
    account_name: Optional[str]
    rows: List[LedgerRow]
    total_debit: float
    total_credit: float
    closing_balance: float

class TrialBalanceRow(BaseModel):
    account_code: str
    account_name: str
    category: str
    debit: float
    credit: float

class TrialBalanceResponse(BaseModel):
    date: datetime
    rows: List[TrialBalanceRow]
    total_debit: float
    total_credit: float

class FinancialReportRow(BaseModel):
    account_name: str
    amount: float

class ProfitLossResponse(BaseModel):
    revenue: List[FinancialReportRow]
    expenses: List[FinancialReportRow]
    total_revenue: float
    total_expenses: float
    net_profit: float

class BalanceSheetResponse(BaseModel):
    assets: List[FinancialReportRow]
    liabilities: List[FinancialReportRow]
    equity: List[FinancialReportRow]
    total_assets: float
    total_liabilities: float
    total_equity: float

File: frontend/src/features/accounts/types/accounts.ts
export type AccountCategory = 'Asset' | 'Liability' | 'Equity' | 'Revenue' | 'Expense';

export interface Account {
  id: string;
  code: string;
  name: string;
  category: AccountCategory;
  is_active: boolean;
  is_system: boolean;
  current_balance: number;
}

export interface JournalEntryLine {
  id?: string;
  account_id: string;
  description?: string;
  debit: number;
  credit: number;
}

export interface JournalEntry {
  id: string;
  reference: string;
  date: string;
  description: string;
  status: 'Draft' | 'Approved' | 'Voided';
  created_at: string;
  lines: JournalEntryLine[];
}

export interface LedgerRow {
  date: string;
  reference: string;
  journal_desc?: string;
  line_desc?: string;
  account_name: string;
  debit: number;
  credit: number;
  balance: number;
  status?: string;
  source_id?: string;
  created_by_name?: string;
}

export interface LedgerResponse {
  account_name?: string;
  rows: LedgerRow[];
  total_debit: number;
  total_credit: number;
  closing_balance: number;
}

export interface TrialBalanceRow {
  account_code: string;
  account_name: string;
  category: string;
  debit: number;
  credit: number;
}

export interface TrialBalanceResponse {
  date: string;
  rows: TrialBalanceRow[];
  total_debit: number;
  total_credit: number;
}

export interface FinancialReportRow {
  account_name: string;
  amount: number;
}

export interface ProfitLossResponse {
  revenue: FinancialReportRow[];
  expenses: FinancialReportRow[];
  total_revenue: number;
  total_expenses: number;
  net_profit: number;
}

export interface BalanceSheetResponse {
  assets: FinancialReportRow[];
  liabilities: FinancialReportRow[];
  equity: FinancialReportRow[];
  total_assets: number;
  total_liabilities: number;
  total_equity: number;
}

export interface DashboardStatsResponse {
  total_revenue: number;
  total_expenses: number;
  net_profit: number;
  total_assets: number;
  cash_balance: number;
  bank_balance: number;
  ar_balance: number;
  ap_balance: number;
}
